examples/mnist/run.py

```diff
@@ -57,43 +64,33 @@ def normalize(image, label):
   def _build_training_graph(self, training_data):
     """Build a graph for plaintext model training."""
 
-    # model parameters and initial values
-    w0 = tf.Variable(tf.random_normal([28 * 28, 512]))
-    b0 = tf.Variable(tf.zeros([512]))
-    w1 = tf.Variable(tf.random_normal([512, 10]))
-    b1 = tf.Variable(tf.zeros([10]))
-
-    # optimizer and data pipeline
-    optimizer = tf.train.AdamOptimizer(learning_rate=0.01)
-
-    # training loop
-    def loop_body(i):
-
-      # get next batch
-      x, y = training_data.get_next()
+    model = keras.Sequential()
+    model.add(keras.layers.Flatten())
+    model.add(keras.layers.Dense(512, input_shape=[28, 28, 1]))
+    model.add(keras.layers.Activation('relu'))
+    model.add(keras.layers.Dense(self.NUM_CLASSES, activation='softmax'))
 
-      # model construction
-      layer0 = tf.matmul(x, w0) + b0
-      layer1 = tf.nn.sigmoid(layer0)
-      layer2 = tf.matmul(layer1, w1) + b1
+    def loss(model, x, y):
+      y_ = model(x)
+      return tf.losses.sparse_softmax_cross_entropy(labels=y, logits=y_)
```

Comment:
this should also go through a `tf.reduce_mean` before being used in the `grad` function below